From 41a0a294b4952857976b6fc40bba52d42b200917 Mon Sep 17 00:00:00 2001
From: Biwen Li <biwen.li@nxp.com>
Date: Thu, 27 Jun 2019 15:02:44 +0800
Subject: [PATCH 003/245] arm64: dts: ls1012a/ls1043a/ls1046a/ls1088a/ls208xa:
 replace ftm0 with ftm_alarm0

commit 198abc32be319c42fc4ce29c00bfc7609777b9d7 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux.git

The patch replaces ftm0 with ftm_alarm0 DT node
	- replace ftm0 with ftm_alarm0
	- add new rcpm node
	- remove old rcpm node
	- aliases ftm_alarm0 as rtc1

Signed-off-by: Biwen Li <biwen.li@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1012a.dtsi | 20 +++++++--------
 .../dts/freescale/fsl-ls1043a-qds-sdk.dts     |  2 +-
 .../dts/freescale/fsl-ls1043a-rdb-sdk.dts     |  2 +-
 .../arm64/boot/dts/freescale/fsl-ls1043a.dtsi | 25 +++++++++++--------
 .../dts/freescale/fsl-ls1046a-frwy-sdk.dts    |  2 +-
 .../dts/freescale/fsl-ls1046a-qds-sdk.dts     |  2 +-
 .../dts/freescale/fsl-ls1046a-rdb-sdk.dts     |  2 +-
 .../arm64/boot/dts/freescale/fsl-ls1046a.dtsi | 24 +++++++++++-------
 .../arm64/boot/dts/freescale/fsl-ls1088a.dtsi | 22 ++++++++++------
 .../arm64/boot/dts/freescale/fsl-ls208xa.dtsi | 14 ++++++++---
 10 files changed, 69 insertions(+), 46 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
index 59bd8c4c7a26..a6a85c3fed74 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1012a.dtsi
@@ -22,6 +22,7 @@
 		rtic-c = &rtic_c;
 		rtic-d = &rtic_d;
 		sec-mon = &sec_mon;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -288,11 +289,16 @@
 			#thermal-sensor-cells = <1>;
 		};
 
-		ftm0: ftm0@29d0000 {
+		rcpm: rcpm@1ee2140 {
+			compatible = "fsl,ls1012a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1ee2140 0x0 0x4>;
+			#fsl,rcpm-wakeup-cells = <1>;
+		};
+
+		ftm_alarm0: timer@29d0000 {
 			compatible = "fsl,ls1012a-ftm-alarm";
-			reg = <0x0 0x29d0000 0x0 0x10000>,
-			      <0x0 0x1ee2140 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
+			reg = <0x0 0x29d0000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x20000>;
 			interrupts = <0 86 0x4>;
 			big-endian;
 		};
@@ -495,12 +501,6 @@
 					<0000 0 0 4 &gic 0 113 IRQ_TYPE_LEVEL_HIGH>;
 			status = "disabled";
 		};
-
-		rcpm: rcpm@1ee2000 {
-			compatible = "fsl,ls1012a-rcpm", "fsl,qoriq-rcpm-2.1";
-			reg = <0x0 0x1ee2000 0x0 0x1000>;
-			fsl,#rcpm-wakeup-cells = <1>;
-		};
 	};
 
 	reserved-memory {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
index 0878d6b8cb6a..43bafb51515e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
@@ -226,7 +226,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts
index 5bfd2262d942..210bc9f25c47 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-sdk.dts
@@ -221,7 +221,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index df45ca9e46d6..af15c8a0dbfb 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -26,6 +26,7 @@
 		ethernet4 = &enet4;
 		ethernet5 = &enet5;
 		ethernet6 = &enet6;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -658,16 +659,6 @@
 			status = "disabled";
 		};
 
-		ftm0: ftm0@29d0000 {
-			compatible = "fsl,ls1043a-ftm-alarm";
-			reg = <0x0 0x29d0000 0x0 0x10000>,
-			      <0x0 0x1ee2140 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
-			interrupts = <0 86 0x4>;
-			big-endian;
-			status = "okay";
-		};
-
 		wdog0: wdog@2ad0000 {
 			compatible = "fsl,ls1043a-wdt", "fsl,imx21-wdt";
 			reg = <0x0 0x2ad0000 0x0 0x10000>;
@@ -868,6 +859,20 @@
 					<0000 0 0 4 &gic 0 157 0x4>;
 			status = "disabled";
 		};
+
+		rcpm: rcpm@1ee2140 {
+			compatible = "fsl,ls1043a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1ee2140 0x0 0x4>;
+			#fsl,rcpm-wakeup-cells = <1>;
+		};
+
+		ftm_alarm0: timer@29d0000 {
+			compatible = "fsl,ls1043a-ftm-alarm";
+			reg = <0x0 0x29d0000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x20000>;
+			interrupts = <0 86 0x4>;
+			big-endian;
+		};
 	};
 
 	firmware {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
index c46a14a091b6..8c0e8aaddc2f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
@@ -196,7 +196,7 @@
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
index 644632275574..d5de7ea92c5b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
@@ -223,7 +223,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
index de4d3b5bf4b1..d9ae081b4f3b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
@@ -228,7 +228,7 @@ pcie@3600000 {
 	dma-coherent;
 };
 
-&ftm0 {
+&ftm_alarm0 {
 	dma-coherent;
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
index cc610bc37905..8e719a76eaf5 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
@@ -27,6 +27,7 @@
 		ethernet5 = &enet5;
 		ethernet6 = &enet6;
 		ethernet7 = &enet7;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -553,15 +554,6 @@
 			status = "disabled";
 		};
 
-		ftm0: ftm0@29d0000 {
-			compatible = "fsl,ls1046a-ftm-alarm";
-			reg = <0x0 0x29d0000 0x0 0x10000>,
-			      <0x0 0x1ee2140 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
-			interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
-			big-endian;
-		};
-
 		wdog0: watchdog@2ad0000 {
 			compatible = "fsl,imx21-wdt";
 			reg = <0x0 0x2ad0000 0x0 0x10000>;
@@ -806,6 +798,20 @@
 			num-lanes = <2>;
 			status = "disabled";
 		};
+
+		rcpm: rcpm@1ee208c {
+			compatible = "fsl,ls1046a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1ee208c 0x0 0x4>;
+			#fsl,rcpm-wakeup-cells = <1>;
+		};
+
+		ftm_alarm0: timer@29d0000 {
+			compatible = "fsl,ls1046a-ftm-alarm";
+			reg = <0x0 0x29d0000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x20000>;
+			interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
+			big-endian;
+		};
 	};
 
 	reserved-memory {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
index 0a72337f990b..e2e02f87a334 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1088a.dtsi
@@ -18,6 +18,7 @@
 
 	aliases {
 		crypto = &crypto;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -449,14 +450,6 @@
 			status = "disabled";
 		};
 
-		ftm0: ftm0@2800000 {
-			compatible = "fsl,ls1088a-ftm-alarm";
-			reg = <0x0 0x2800000 0x0 0x10000>,
-			      <0x0 0x1e34050 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
-			interrupts = <0 44 4>;
-		};
-
 		i2c0: i2c@2000000 {
 			compatible = "fsl,vf610-i2c", "fsl,ls1088a-vf610-i2c";
 			#address-cells = <1>;
@@ -905,6 +898,19 @@
 				     <0 208 4>,
 				     <0 209 4>;
 		};
+
+		rcpm: rcpm@1e34040 {
+			compatible = "fsl,ls1088a-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1e34040 0x0 0x18>;
+			#fsl,rcpm-wakeup-cells = <6>;
+		};
+
+		ftm_alarm0: timer@2800000 {
+			compatible = "fsl,ls1088a-ftm-alarm";
+			reg = <0x0 0x2800000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x0 0x0 0x0 0x0 0x4000 0x0>;
+			interrupts = <0 44 4>;
+		};
 	};
 
 	firmware {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
index 59eb777ae202..cde34c161e1a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
@@ -22,6 +22,7 @@
 		crypto = &crypto;
 		serial0 = &serial0;
 		serial1 = &serial1;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpu: cpus {
@@ -731,11 +732,16 @@
 			interrupts = <0 12 4>;
 		};
 
-		ftm0: ftm0@2800000 {
+		rcpm: rcpm@1e34040 {
+			compatible = "fsl,ls208xa-rcpm", "fsl,qoriq-rcpm-2.1+";
+			reg = <0x0 0x1e34040 0x0 0x18>;
+			#fsl,rcpm-wakeup-cells = <6>;
+		};
+
+		ftm_alarm0: timer@2800000 {
 			compatible = "fsl,ls208xa-ftm-alarm";
-			reg = <0x0 0x2800000 0x0 0x10000>,
-			      <0x0 0x1e34050 0x0 0x4>;
-			reg-names = "ftm", "pmctrl";
+			reg = <0x0 0x2800000 0x0 0x10000>;
+			fsl,rcpm-wakeup = <&rcpm 0x0 0x0 0x0 0x0 0x4000 0x0>;
 			interrupts = <0 44 4>;
 		};
 	};
-- 
2.17.1

